Broken pytest for `process_images`

```
=================================== FAILURES ===================================
[24](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:24)
______________________________ test_get_spectrum _______________________________
[25](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:25)

[26](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:26)
im = dask.array<mean_agg-aggregate, shape=(516, 516), dtype=float64, chunksize=(516, 516), chunktype=numpy.ndarray>
[27](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:27)

[28](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:28)
    def test_get_spectrum(im):
[29](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:29)
>       spectrum = process_images.get_spectrum(
[30](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:30)
            im, [-8.28334263e-05, 1.33928571e-02, 1.64000000e+02], biny=2
[31](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:31)
        )
[32](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:32)

[33](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:33)
polartools/tests/test_process_images.py:95: 
[34](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:34)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[35](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:35)
polartools/process_images.py:261: in get_spectrum
[36](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:36)
    return extract(ph, curvature, biny=biny)
[37](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:37)
polartools/_pyrixs.py:176: in extract
[38](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:38)
    np.nanmin(corrected_y), np.nanmax(corrected_y), biny
[39](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:39)
<__array_function__ internals>:5: in nanmin
[40](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:40)
    ???
[41](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:41)
/usr/share/miniconda/lib/python3.9/site-packages/dask/array/core.py:1681: in __array_function__
[42](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:42)
    return da_func(*args, **kwargs)
[43](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:43)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[44](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:44)

[45](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:45)
a = dask.array<sub, shape=(nan,), dtype=float64, chunksize=(nan,), chunktype=numpy.ndarray>
[46](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:46)
axis = None, keepdims = False, split_every = None, out = None
[47](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:47)

[48](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:48)
    @derived_from(np)
[49](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:49)
    def nanmin(a, axis=None, keepdims=False, split_every=None, out=None):
[50](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:50)
        if np.isnan(a.size):
[101](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:101)
/usr/share/miniconda/lib/python3.9/site-packages/dask/array/reductions.py:521: ValueError
[102](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:102)
=============================== warnings summary ===============================
[103](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:103)
../../../../../usr/share/miniconda/lib/python3.9/site-packages/pims/cine.py:29
[104](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:104)
  /usr/share/miniconda/lib/python3.9/site-packages/pims/cine.py:29: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.10 it will stop working
[105](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:105)
    from collections import Iterable
[106](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:106)

[107](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:107)
../../../../../usr/share/miniconda/lib/python3.9/site-packages/mongoquery/__init__.py:7
[108](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:108)
../../../../../usr/share/miniconda/lib/python3.9/site-packages/mongoquery/__init__.py:7
[109](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:109)
  /usr/share/miniconda/lib/python3.9/site-packages/mongoquery/__init__.py:7: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.10 it will stop working
[110](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:110)
    from collections import Sequence, Mapping
[111](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:111)

[112](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:112)
polartools/tests/test_manage_database.py::test_manage_databroker_database
[113](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:113)
  /usr/share/miniconda/lib/python3.9/site-packages/databroker/core.py:1116: UserWarning: The method canonical has been renamed documents. This alias may be removed in a future release.
[114](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:114)
    warnings.warn(
[115](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:115)

[116](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:116)
-- Docs: https://docs.pytest.org/en/stable/warnings.html
[117](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:117)

[118](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:118)
----------- coverage: platform linux, python 3.9.7-final-0 -----------
[119](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:119)
Coverage XML written to file coverage.xml
[120](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:120)

[121](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:121)
=========================== short test summary info ============================
[122](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:122)
FAILED polartools/tests/test_process_images.py::test_get_spectrum - ValueErro...
[123](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:123)
FAILED polartools/tests/test_process_images.py::test_get_spectra - ValueError...
[124](https://github.com/APS-4ID-POLAR/polartools/runs/5827832219?check_suite_focus=true#step:8:124)
================== 2 failed, 22 passed, 4 warnings in 40.08s ===================
```

--- a/polartools/process_images.py
+++ b/polartools/process_images.py
@@ -250,13 +250,15 @@ def get_spectrum(image, curvature, biny=1):
 
     Returns
     -------
-    spectrum : same format as image input
+    spectrum : numpy array
         Extracted 1D spectrum.
 
     See also
     --------
     :func:`pyrixs.extract`
     """
+    if isinstance(image, da.core.Array):
+        image = image.compute()
     ph = _cleanup_photon_events(image_to_photon_events(image.transpose()))
     return extract(ph, curvature, biny=biny)
 
@@ -293,6 +295,8 @@ def get_spectra(images, curvature, biny=1):
     """
     spectra = []
     for image in images:
+        if isinstance(image, da.core.Array):
+            image = image.compute()
         spectra.append(get_spectrum(image, curvature, biny=biny))
     return np.array(spectra)
 

--- a/polartools/tests/test_process_images.py
+++ b/polartools/tests/test_process_images.py
@@ -97,7 +97,7 @@ def test_get_spectrum(im):
     )
 
     assert spectrum.shape == (264, 2)
-    assert allclose(spectrum.mean().compute(), 132.42162878787877)
+    assert allclose(spectrum.mean(), 132.42162878787877)
 
 
 def test_get_spectra(ims):

